include/libopencm3/stm32/l1/ri.h

```diff
@@ -0,0 +1,658 @@
+ *
+ * LGPL License Terms @ref lgpl_license
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2016 Marek Koza <qyx@krtko.org>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Based on the RM0038 Reference manual
+ * (STM32L100xx, STM32L151xx, STM32L152xx and STM32L162xx advanced ARM®-based
+ * 32-bit MCUs)
+ */
+/**@{*/
+
+/*
+ * There is a mismatch in the RM0038 in the RI register offset addresses (they
+ * are relative to the COMP base address instead of the RI base address for an
+ * unknown reason). We are subtracting 4 in order to use them with ROUTING_BASE.
+ */
+#define RI_BASE                 ROUTING_BASE - 0x04
+
+#define RI_ICR                  MMIO32(RI_BASE + 0x04)
+#define RI_ASCR1                MMIO32(RI_BASE + 0x08)
+#define RI_ASCR2                MMIO32(RI_BASE + 0x0c)
+#define RI_HYSCR1               MMIO32(RI_BASE + 0x10)
+#define RI_HYSCR2               MMIO32(RI_BASE + 0x14)
+#define RI_HYSCR3               MMIO32(RI_BASE + 0x18)
+#define RI_HYSCR4               MMIO32(RI_BASE + 0x1c)
+#define RI_ASMR1                MMIO32(RI_BASE + 0x20)
+#define RI_CMR1                 MMIO32(RI_BASE + 0x24)
+#define RI_CICR1                MMIO32(RI_BASE + 0x28)
+#define RI_ASMR2                MMIO32(RI_BASE + 0x2c)
+#define RI_CMR2                 MMIO32(RI_BASE + 0x30)
+#define RI_CICR2                MMIO32(RI_BASE + 0x34)
+#define RI_ASMR3                MMIO32(RI_BASE + 0x38)
+#define RI_CMR3                 MMIO32(RI_BASE + 0x3c)
+#define RI_CICR3                MMIO32(RI_BASE + 0x40)
+#define RI_ASMR4                MMIO32(RI_BASE + 0x44)
+#define RI_CMR4                 MMIO32(RI_BASE + 0x48)
+#define RI_CICR4                MMIO32(RI_BASE + 0x4c)
+#define RI_ASMR5                MMIO32(RI_BASE + 0x50)
+#define RI_CMR5                 MMIO32(RI_BASE + 0x54)
+#define RI_CICR5                MMIO32(RI_BASE + 0x58)
+
+/**
+ * RI input capture register
+ *
+ * The RI_ICR register is used to select the routing of 4 full ports to the
+ * input captures of TIM2, TIM3 and TIM4.
+ */
+#define RI_ICR_IC1IOS_SHIFT     0
+#define RI_ICR_IC1IOS_MASK      0xf
+#define RI_ICR_IC2IOS_SHIFT     4
+#define RI_ICR_IC2IOS_MASK      0xf
+#define RI_ICR_IC3IOS_SHIFT     8
+#define RI_ICR_IC3IOS_MASK      0xf
+#define RI_ICR_IC4IOS_SHIFT     12
+#define RI_ICR_IC4IOS_MASK      0xf
```

Comment:
just thinking out loud, how about RI_ICR_ICxIOS_SHIFT(x)  and RI_ICR_ICxIOS_MASK == 0xf rather than duplicating them? maybe not worth it.